src/python/pants/option/arg_splitter.py

```diff
@@ -204,7 +211,7 @@ def spec(arg: str) -> bool:
     An arg is a spec if it looks like an AddressSpec or a FilesystemSpec.
     In the future we can expand this heuristic to support other kinds of specs, such as URLs.
     """
-    return os.path.sep in arg or ':' in arg or os.path.isdir(arg)
+    return os.path.sep in arg or ':' in arg or Path(arg).exists()
```

Comment:
Note that the statement "the path must exist to be considered a spec" is only true for top-level files/dirs. Anything else will have at least one `os.path.sep` in it, so will be read as a spec:

`./pants path/to/nonexistent/file` will totally work, because of the first condition.

Reply:
We should probably check that realpath doesn't contain any `..` anywhere, so it doesn't escape the buildroot.

Reply:
CLI spec normalization handles this for addresses, IIRC. And the engine won't allow PathGlobs to escape the root.

Reply:
Correct on PathGlobs not allowing escaping the root. Thanks for pointing to this Stu!

```
▶ ./pants list ..
Scrubbed PYTHONPATH=/Users/eric/DocsLocal/code/projects/pants/src/python: from the environment.
timestamp: 2020-01-09T22:55:06.434113
Exception caught: (pants.engine.scheduler.ExecutionError)
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/pants_loader.py", line 85, in <module>
    main()
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/pants_loader.py", line 81, in main
    PantsLoader.run()
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/pants_loader.py", line 77, in run
    cls.load_and_execute(entrypoint)
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/pants_loader.py", line 70, in load_and_execute
    entrypoint_main()
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/pants_exe.py", line 32, in main
    PantsRunner(start_time=start_time).run()
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/pants_runner.py", line 106, in run
    return runner.run()
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/local_pants_runner.py", line 253, in run
    self._run()
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/local_pants_runner.py", line 331, in _run
    engine_result = self._maybe_run_v2()
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/local_pants_runner.py", line 301, in _maybe_run_v2
    self._target_roots,
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/init/engine_initializer.py", line 225, in run_goal_rules
    exit_code = self.scheduler_session.run_goal_rule(goal_product, params)
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/engine/scheduler.py", line 471, in run_goal_rule
    self._trace_on_error([exc], request)
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/engine/scheduler.py", line 449, in _trace_on_error
    unique_exceptions,

Exception message: 1 Exception encountered:
Computing Select((<pants.engine.console.Console object at 0x1b91fd536d8>+AddressSpecs(dependencies=(SingleAddress(directory='..', name='..'),), matcher=AddressSpecsMatcher(tags=(), exclude_patterns=()))+OptionsBootstrapper(env_tuples=(('PANTS_DEV', '1'),), bootstrap_args=(), args=('/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/pants_loader.py', 'list', '..'), config=_ChainedConfig(chained_configs=(<pants.option.config._SingleFileConfig object at 0x1122c4fd0>,)))), List)
  Computing Task(list_targets(), (<pants.engine.console.Console object at 0x1b91fd536d8>+AddressSpecs(dependencies=(SingleAddress(directory='..', name='..'),), matcher=AddressSpecsMatcher(tags=(), exclude_patterns=()))+OptionsBootstrapper(env_tuples=(('PANTS_DEV', '1'),), bootstrap_args=(), args=('/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/pants_loader.py', 'list', '..'), config=_ChainedConfig(chained_configs=(<pants.option.config._SingleFileConfig object at 0x1122c4fd0>,)))), List, false)
    Computing Task(remove_provenance(), AddressSpecs(dependencies=(SingleAddress(directory='..', name='..'),), matcher=AddressSpecsMatcher(tags=(), exclude_patterns=())), BuildFileAddresses, true)
      Computing Task(provenanced_addresses_from_address_families(), AddressSpecs(dependencies=(SingleAddress(directory='..', name='..'),), matcher=AddressSpecsMatcher(tags=(), exclude_patterns=())), ProvenancedBuildFileAddresses, true)
        Computing Snapshot(PathGlobs(include=('../BUILD', '../BUILD.*'), exclude=('.*/', 'bower_components/', 'node_modules/', '*.egg-info/'), glob_match_error_behavior=<GlobMatchErrorBehavior.ignore: 'ignore'>, conjunction=<GlobExpansionConjunction.any_match: 'any_match'>))
          Throw(Failed to parse PathGlobs: Failed to parse PathGlobs for include(["../BUILD", "../BUILD.*"]), exclude([".*/", "bower_components/", "node_modules/", "*.egg-info/"]): Globs may not traverse outside of the buildroot: "../BUILD")
            Traceback (no traceback):
              <pants native internals>
            Exception: Failed to parse PathGlobs: Failed to parse PathGlobs for include(["../BUILD", "../BUILD.*"]), exclude([".*/", "bower_components/", "node_modules/", "*.egg-info/"]): Globs may not traverse outside of the buildroot: "../BUILD"
```